x/ecocredit/client/tx.go

```diff
@@ -107,7 +107,7 @@ Parameters:
 			if args[2] == "" {
 				return errors.New("base64_metadata is required")
 			}
-			b, err := decodeMetadata(args[2])
+			b, err := types.DecodeMetadata(args[2])
```

Comment:
why do we need to decode the metadata now?
it can just be provided as any type of string and doesn't need to be decoded since it'll be stored as string too

Reply:
This was a misunderstanding on my part and it was not clear in the issue. I was under the impression that we wanted to display the decoded metadata and was not aware that the intention was to support any type of string as well. The issue was proposed when Mark was asking about which credit class was which and shared class query results.

For example...

```
class:
  ...
  metadata: cmVnZW4=
```

Would be...

```
class:
  ...
  metadata: regen
```

Do you think it would still be valuable to decode the metadata if base64 encoded string and return string otherwise? Or are you proposing that we display the base64 encoded string? cc @blushi @aaronc

Reply:
I believe we shouldn't have any based64 encoding/decoding involved anymore now that we just deal with strings 
just as for the metadata fields in x/group and x/gov

Reply:
so the changes in this PR just affect the new V1/ORM stuff, and so far none of the client CLI stuff has been updated for this (CLI still targeting v1alpha2). i'll open a tracking issue for this

Reply:
https://github.com/regen-network/regen-ledger/issues/876

Reply:
oh ok I see

Reply:
i will update the core however to get rid of decoding, i also had thought the point was to keep base64 strings at the storage level, and decode at the query level. will update shortly

Reply:
yeah indeed we don't need the decoding in the queries